Editor/MissingReferences/MissingProjectReferences.cs

```diff
@@ -0,0 +1,276 @@
+            /// </summary>
+            class AssetContainer
+            {
+                readonly UnityObject m_Object;
+                public readonly List<SerializedProperty> PropertiesWithMissingReferences = new List<SerializedProperty>();
+
+                public UnityObject UnityObject { get { return m_Object; } }
+
+                /// <summary>
+                /// Initialize an AssetContainer to represent the given UnityObject
+                /// This will scan the object for missing references and retain the information for display in
+                /// the given window.
+                /// </summary>
+                /// <param name="unityObject">The UnityObject to scan for missing references</param>
+                /// <param name="options">User-configurable options for this view</param>
+                public AssetContainer(UnityObject unityObject, Options options)
+                {
+                    m_Object = unityObject;
+                    CheckForMissingReferences(unityObject, PropertiesWithMissingReferences, options);
+                }
+
+                /// <summary>
+                /// Draw the missing references UI for this asset
+                /// </summary>
+                public void Draw()
+                {
+                    using (new EditorGUI.IndentLevelScope())
+                    {
+                        DrawPropertiesWithMissingReferences(PropertiesWithMissingReferences);
+                    }
+                }
+            }
+
+            readonly SortedDictionary<string, Folder> m_Subfolders = new SortedDictionary<string, Folder>();
+            readonly List<GameObjectContainer> m_Prefabs = new List<GameObjectContainer>();
+            readonly List<AssetContainer> m_Assets = new List<AssetContainer>();
+            bool m_Visible;
+
+            /// <summary>
+            /// The number of assets in this folder with missing references
+            /// </summary>
+            public int Count;
+
+            /// <summary>
+            /// Clear the contents of this container
+            /// </summary>
+            public void Clear()
+            {
+                m_Subfolders.Clear();
+                m_Prefabs.Clear();
+                m_Assets.Clear();
+                Count = 0;
+            }
+
+            /// <summary>
+            /// Scan the contents of a given path and add the results as a subfolder to this container
+            /// </summary>
+            /// <param name="path">The path to scan</param>
+            /// <param name="options">User-configurable options for this view</param>
+            public void AddAssetAtPath(string path, Options options)
+            {
+                var prefab = AssetDatabase.LoadAssetAtPath<GameObject>(path);
+                if (prefab != null)
+                {
+                    if (PrefabUtility.GetPrefabAssetType(prefab) == PrefabAssetType.Model)
```

Comment:
I think it *might* be possible this asset can have missing references if the textures have been set to external and deleted

Reply:
Oh interesting... I figured anything in a model that was missing would be internal references. Can you help me set this up so we can test it?

Reply:
Good catch! I dug up our friend the horse from UseCaseContent, which has an embedded material and texture.

I needed to add sub-assets in order to catch the missing texture reference on the embedded material, which also means we'll be catching a lot more edge cases of sub-assets with missing references. 🤘 

The other thing I've done is to introduce a base class so that we can put AssetContainers and GameObjectContainers in the same list. I needed this for the sub-assets to be a single list, and it also means that we don't end up sorting prefabs and other assets separately. I forgot to take care of this in my pre-PR cleanup pass so I'm glad you gave me a reason to revisit.